swift/ql/lib/codeql/swift/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -97,6 +120,10 @@ private class SsaDefinitionNodeImpl extends SsaDefinitionNode, NodeImpl {
   override DataFlowCallable getEnclosingCallable() {
     result = TDataFlowFunc(def.getBasicBlock().getScope())
   }
+
+  override DataFlowType getTypeImpl() {
+    result = getDataFlowType(this.asDefinition().getSourceVariable().asVarDecl().getType())
```

Comment:
Not all `SourceVariable`s has a result for `asVarDecl` (in particular, some of them are related to key paths: https://github.com/github/codeql/blob/main/swift/ql/lib/codeql/swift/dataflow/Ssa.qll#L28).

It seems like these don't have a type now? Should we instead have a `getType` predicate on `SourceVariable` (that _all_ instances of the abstract class then implements) so that this predicate could instead be implemented as `getDataFlowType(this.asDefinition().getSourceVariable())`?

Reply:
Done.